### PR TITLE
PRIM-242BillingandInvoice

### PR DESCRIPTION
**Pull request by bhatt7821**
PRIM-242BillingandInvoice

### CODE REVIEW

File: Tests/Steps/Add_Edit_Tenant_User_StepDefination.cs [hunk 1]
```diff
@@ -0,0 +1,230 @@
+﻿using Affordit_Automation.PageObjects;
```
Comment (ahaynes91) [inline Tests/Steps/Add_Edit_Tenant_User_StepDefination.cs:1]: First comment applies to this file as well.

File: Tests/Steps/BillingAndInvoiceStepDefinition.cs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    public sealed class BillingAndInvoiceStepDefinition
+    {
+        private IWebDriver _driver;
+
+        Actions act;
+        WebDriverWait wait;
+
+        protected readonly PropertyReader _propertyReader;
+        protected readonly BillingAndInvoicePageObjects _billingAndInvoicePageObjects;
+        // protected string loggedUser;
```
Comment (ahaynes91) [inline Tests/Steps/BillingAndInvoiceStepDefinition.cs:25]: Please remove

File: Tests/Steps/BillingAndInvoiceStepDefinition.cs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+       
+        private readonly ScenarioContext _scenarioContext;
+
+        public BillingAndInvoiceStepDefinition(IWebDriver driver)
+        {
+            _driver = driver;
+            _propertyReader = PropertyReader.Instance;
+            _billingAndInvoicePageObjects = new BillingAndInvoicePageObjects(_driver);
+            act = new Actions(driver);
```
Comment (ahaynes91) [inline Tests/Steps/BillingAndInvoiceStepDefinition.cs:34]: _driver

**Review by ahaynes91 — CHANGES_REQUESTED** [review]

File: PageObjects/Add_Edit_Tenant_User_PageObjects.cs [hunk 4]
```diff
@@ -0,0 +1,221 @@
+﻿using Affordit_Automation.Utils;
```
[the remark this answers is not in the document]
Comment (bhatt7821) [reply PageObjects/Add_Edit_Tenant_User_PageObjects.cs:1]: ok

File: Tests/Steps/Add_Edit_Tenant_User_StepDefination.cs (hunk 1 above)
Comment (bhatt7821) [reply Tests/Steps/Add_Edit_Tenant_User_StepDefination.cs:1]: ok

File: Tests/Steps/BillingAndInvoiceStepDefinition.cs (hunk 2 above)
Comment (bhatt7821) [reply Tests/Steps/BillingAndInvoiceStepDefinition.cs:25]: ok

File: Tests/Steps/BillingAndInvoiceStepDefinition.cs (hunk 3 above)
Comment (bhatt7821) [reply Tests/Steps/BillingAndInvoiceStepDefinition.cs:34]: _driver is an object of IWebDriver

**Review by ahaynes91 — APPROVED** [review]

**Pull request merged**